spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdSink.java

```diff
@@ -21,7 +22,7 @@
 
     public StatsdSink(Properties properties, MetricRegistry registry, org.apache.spark.SecurityManager securityMgr) {
         logger.info("Starting StatsdSink with properties:\n" + properties.toString());
-        this.prefix = properties.getProperty(Keys.PREFIX, Defaults.PREFIX);
+        this.prefix = MetricRegistry.name(MANDATORY_PREFIX, properties.getProperty(Keys.PREFIX, Defaults.PREFIX));
```

Comment:
I don't think that user-provided prefix should go after the static one. Example:
* regular metric name with static prefix: `spark.jvm.heap_used`
* user sets a variable named "prefix"
* user seems to expect it to actually prefix the metric name: `dev.spark.jvm.heap_used` or `prod.spark.jvm.heap_used`

Static prefix goes first only when there's no user-provided prefix. Otherwise, user-provided prefix prefixes the static one.

Reply:
changed as suggested